python/feathub/processors/local/ast_evaluator/local_func_evaluator.py

```diff
@@ -12,11 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from datetime import tzinfo, timezone
-from typing import Any
+from typing import Any, Union, Optional
 
 from feathub.common.utils import to_unix_timestamp
 
 
+def local_processor_to_unix_timestamp(
+    time: Union[str, None],
+    format: str = "%Y-%m-%d %H:%M:%S",
+    tz: tzinfo = timezone.utc,
+) -> Optional[int]:
+    if time is None:
+        return None
+    try:
+        return int(to_unix_timestamp(time, format=format, tz=tz))
+    except ValueError as v:
+        # Strip off the unconverted part of the given time in str. The behavior is
+        # consistent with FlinkProcessor and SparkProcessor
```

Comment:
Can you explain where this logic is implemented for FlinkProcessor and SparkProcessor?

And if we want to enforce this logic across all processors, would it be simpler to do it in `to_unix_timestamp`?

Reply:
After running some tests, It turns out that this behavior is the same as FlinkProcessor but different from SparkProcessor. In FlinkProcessor the unconverted part is dropped, while the Spark job will fail if there are unconverted parts.

I think Feathub should just pick one of them. And we need to implement the behavior for the other processor using UDF. I think the behavior in Flink is more robust. What do you think?

Reply:
Since we want to consistently dropped the uncovered part, would it be simpler to implement this logic in `feathub.common.utils.to_unix_timestamp`?

Reply:
Thanks for the explanation. I also find it to be hacky to use the error message format. I agree we should go with the Spark behavior. And I added a TODO to FlinkProcessor to add a UDF to make the behavior consistent.